web/modeltr/document.py

```diff
@@ -126,15 +126,21 @@ def __insert(self, **kwargs):
         returning_id = cur.fetchone()[0]
         self.id = str(returning_id)
 
-    def to_dict(self):
+    def to_dict(self, redacted=None):
         result = {}
         for prop, typ in self.__types.items():
             if prop == 'id':
                 continue
             if isinstance(getattr(self, prop), type(datetime.datetime.now())):
                 result.update({prop: getattr(self, prop).strftime(self.__datetime_format)})
             else:
-                result.update({prop: getattr(self, prop)})
+                if redacted:
+                    if len(str(getattr(self, prop))) > 20:
+                        result.update({prop: '{}... redacted'.format(str(getattr(self, prop))[:20])})
+                    else:
+                        result.update({prop: getattr(self, prop)})
+                else:
+                    result.update({prop: getattr(self, prop)})
```

Comment:
I would suggest to change this block to something like:
```
MAXIMUM_VALUE_LENGTH = 100 # Limit in order to prevent excessively long data, such as base 64
value_length = len(str(getattr(self, prop)))

if redacted and value_length > MAXIMUM_VALUE_LENGTH:
    result.update({prop: '{}... redacted'.format(str(getattr(self, prop))[:MAXIMUM_VALUE_LENGTH])})
else:
    result.update({prop: getattr(self, prop)})
```

Reply:
ok